internal/services/sources_service.go

```diff
@@ -47,3 +49,39 @@ func GetSource(w http.ResponseWriter, r *http.Request) {
 		renderError(w, r, payloads.NewRenderError(r.Context(), "show source", err))
 	}
 }
+
+func fetchARN(ctx context.Context, client sources.SourcesIntegration, sourceId string) (string, error) {
+	// Get all the authentications linked to a specific source
+	resp, err := client.ListSourceAuthenticationsWithResponse(ctx, sourceId, &sources.ListSourceAuthenticationsParams{}, AddIdentityHeader)
+	if err != nil {
+		return "", fmt.Errorf("cannot list source authentication: %w", err)
+	}
+
+	// Filter authentications to include only auth where resource_type == "Application"
+	// TODO: use sources client filtering after they solve their bug
```

Comment:
Can we remove TODO from here and create an issue in our project that will reference their issue? From my experience, this TODO will likely stays forever if the solution is good enough :-) My IDE warns about TODO comments on every commit, that is the reason.